### PR TITLE
feat(cf): Share/Unshare services

### PR DESCRIPTION
**Pull request by stuart-pollock**
spinnaker/spinnaker#4065

### CODE REVIEW

**Comment by stuart-pollock** [comment]
Screen shots showing the layouts of the pipeline stages for Share and Unshare.

Screen Shot 2019-03-13 at 5 26 44 PM
Screen Shot 2019-03-13 at 5 26 27 PM

File: app/scripts/modules/core/src/pipeline/config/stages/shareService/shareServiceStage.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { module } from 'angular';
+
+import { Registry } from 'core/registry';
+import { ExecutionDetailsTasks } from '../common';
+import { ShareServiceExecutionDetails } from './ShareServiceExecutionDetails';
+
+export const SHARE_SERVICE_STAGE = 'spinnaker.core.pipeline.stage.shareService';
+
+module(SHARE_SERVICE_STAGE, []).config(() => {
```
Comment (jkschneider) [inline app/scripts/modules/core/src/pipeline/config/stages/shareService/shareServiceStage.ts]: Look to `waitStage` for an example of why we don't need to wrap this in an angular component anymore.

File: app/scripts/modules/cloudfoundry/src/pipeline/stages/unshareService/cloudfoundryUnshareServiceStage.module.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { ExecutionDetailsTasks, IStage, Registry } from '@spinnaker/core';
+import { CloudfoundryServiceExecutionDetails } from 'cloudfoundry/presentation';
+
+class CloudFoundryUnshareServiceStageCtrl implements IController {
+  public static $inject = ['$scope'];
+  constructor(public $scope: IScope) {}
+}
+
+export const CLOUD_FOUNDRY_UNSHARE_SERVICE_STAGE = 'spinnaker.cloudfoundry.pipeline.stage.unshareServiceStage';
+module(CLOUD_FOUNDRY_UNSHARE_SERVICE_STAGE, [])
```
Comment (jkschneider) [inline app/scripts/modules/cloudfoundry/src/pipeline/stages/unshareService/cloudfoundryUnshareServiceStage.module.ts]: Look to `waitStage` for an example of why we don't need to wrap this in an angular component anymore.

File: app/scripts/modules/core/src/pipeline/config/stages/shareService/shareServiceStage.ts (hunk 1 above)
Comment (stuart-pollock) [reply app/scripts/modules/core/src/pipeline/config/stages/shareService/shareServiceStage.ts]: 👍

File: app/scripts/modules/cloudfoundry/src/pipeline/stages/unshareService/cloudfoundryUnshareServiceStage.module.ts (hunk 2 above)
Comment (stuart-pollock) [reply app/scripts/modules/cloudfoundry/src/pipeline/stages/unshareService/cloudfoundryUnshareServiceStage.module.ts]: 👍

File: app/scripts/modules/cloudfoundry/src/pipeline/stages/shareService/CloudfoundryShareServiceStageConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+      this.setState({ regions });
+      if (region) {
+        this.clearAndResetShareToRegionList(region, regions);
+      }
+    });
+  };
+
+  private serviceInstanceNameUpdated = (event: React.ChangeEvent<HTMLInputElement>): void => {
+    const serviceInstanceName = event.target.value;
+    this.setState({ serviceInstanceName });
```
Comment (jkschneider) [inline app/scripts/modules/cloudfoundry/src/pipeline/stages/shareService/CloudfoundryShareServiceStageConfig.tsx]: We can make this fully controlled.

**Pull request merged**